hardware/raspberrypi/revision-codes/README.md

```diff
@@ -0,0 +1,94 @@
+| 0013 | B+    | 1.2      | 512 MB          | ?            |
+| 0014 | CM1   | 1.0      | 512 MB          | Embest       |
+| 0015 | A+    | 1.1      | 256 MB / 512 MB | Embest       |
+
+## New-style revision codes
+
+Since the Raspberry Pi 2 launch, new-style revision codes were introduced. Rather than being sequential, each bit of the hex code represents a piece of information about the revision:
+
+```
+uuuuuuuuFMMMCCCCPPPPTTTTTTTTRRRR
+```
+
+| Part     | Represents   | Options               |
+| -------- | ------------ | --------------------- |
+| uuuuuuuu | Unused       | Unused                |
+| F        | New flag     | 1: new-style revision |
+|          |              | 0: old-style revision |
+| MMM      | Memory size  | 0: 256 MB             |
+|          |              | 1: 512 MB             |
+|          |              | 2: 1 GB               |
+| CCCC     | Manufacturer | 0: Sony UK            |
+|          |              | 1: Egoman             |
+|          |              | 2: Embest             |
+|          |              | 3: Sony Japan         |
+| PPPP     | Processor    | 0: BCM2835            |
+|          |              | 1: BCM2836            |
+|          |              | 2: BCM2837            |
+| TTTTTTTT | Type         | 0: A                  |
+|          |              | 1: B                  |
+|          |              | 2: A+                 |
+|          |              | 3: B+                 |
+|          |              | 4: 2B                 |
+|          |              | 5: Alpha (??)         |
+|          |              | 6: CM1                |
+|          |              | 8: 3B                 |
+|          |              | 9: Zero               |
+|          |              | a: CM3                |
+|          |              | c: Zero W             |
+| RRRR     | Revision     | 0, 1, 2, etc.         |
+
+New-style revision codes in use:
+
+| Code   | Model             | Revision | RAM    | Manufacturer |
+| ------ | ----------------- | -------- | -------| ------------ |
+| a01040 | 2B                | 1.0      | 1 GB   | Sony UK      |
+| a01041 | 2B                | 1.1      | 1 GB   | Sony UK      |
+| a21041 | 2B                | 1.1      | 1 GB   | Embest       |
+| a22042 | 2B (with BCM2837) | 1.2      | 1 GB   | Embest       |
+| 900021 | A+                | 1.1      | 512 MB | Sony UK      |
+| 900032 | B+                | 1.2      | 512 MB | Sony UK      |
+| 900092 | Zero              | 1.2      | 512 MB | Sony UK      |
+| 900093 | Zero              | 1.3      | 512 MB | Sony UK      |
+| 920093 | Zero              | 1.3      | 512 MB | Embest       |
+| 9000c1 | Zero W            | 1.1      | 512 MB | Sony UK      |
+| a02082 | 3B                | 1.2      | 1 GB   | Sony UK      |
+| a020a0 | CM3               | 1.0      | 1 GB   | Sony UK      |
+| a22082 | 3B                | 1.2      | 1 GB   | Embest       |
+| a32082 | 3B                | 1.2      | 1 GB   | Sony Japan   |
```

Comment:
Like the previous table, I wonder if this table should be sorted numerically by the `Code` column?